packages/@aws-cdk/aws-cloudfront/test/test.basic.ts

```diff
@@ -325,4 +327,199 @@ export = {
     test.done();
   },
 
+  'throws if certificate is not in us-east-1 - fromCertificateArn'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const sourceBucket = new s3.Bucket(stack, 'Bucket');
+    const certificate = certificatemanager.Certificate.fromCertificateArn(
+        stack, 'EuCertificate', 'arn:aws:acm:eu-west-1:1234567890:certificate/testACM'
+    );
+
+    // WHEN
+    const toThrow = () => {
+      new CloudFrontWebDistribution(stack, 'AnAmazingWebsiteProbably', {
+          originConfigs: [
+          {
+            s3OriginSource: {s3BucketSource: sourceBucket},
+            behaviors: [{isDefaultBehavior: true}]
+          }
+        ],
+        aliasConfiguration: {
+          names: ['www.example.com'],
+          acmCertRef: certificate.certificateArn
+        }
+      });
+    };
+
+    // THEN
+    test.throws(() => toThrow(), /acmCertificateArn must be in the 'us-east-1' region, got 'eu-west-1'/);
+    test.done();
+  },
+
+  'throws if stack region is invalid - constructor'(test: Test) {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'RegionStack', {env: {region: 'eu-west-3'}});
+    const sourceBucket = new s3.Bucket(stack, 'Bucket');
+    const certificate = new certificatemanager.Certificate(stack, 'TestCertificate', {
+      domainName: 'www.example.com',
+    });
+
+    // WHEN
+    const toThrow = () => {
+      new CloudFrontWebDistribution(stack, 'AnAmazingWebsiteProbably', {
+        originConfigs: [
+          {
+            s3OriginSource: {s3BucketSource: sourceBucket},
+            behaviors: [{isDefaultBehavior: true}]
+          }
+        ],
+        aliasConfiguration: {
+          names: ['www.example.com'],
+          acmCertRef: certificate.certificateArn
+        }
+      });
+    };
+
+    // THEN
+    test.throws(() => toThrow(), /acmCertificateArn must be in the 'us-east-1' region, got 'eu-west-3'/);
+    test.done();
+  },
+
+  'throws if certificate region is invalid - constructor'(test: Test) {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'RegionStack', {env: {region: 'us-east-1'}});
+    const sourceBucket = new s3.Bucket(stack, 'Bucket');
+    const hostedZone = route53.HostedZone.fromHostedZoneAttributes(stack, 'zone', {
+      zoneName: 'zoneName',
+      hostedZoneId: 'hostedZoneId'
+    });
+    const certificate = new certificatemanager.DnsValidatedCertificate(stack, 'TestCertificate', {
+      domainName: 'www.example.com',
+      hostedZone,
+      region: 'eu-west-3',
```

Comment:
I should be able to resolve that region, right? It's currently resolved as `${Token[Default.Arn.132]}`

Reply:
That makes this PR kind of useless though. The main use was to not request an unusable certificate. If we already have the ARN for it, it's too late.

Would you be open to deprecating `acmCertRef` for a `acmCert: Certificate` property, to be able to access the `region` property?

Reply:
Yes

Reply:
I just noticed that this will cause a typing issue. `acmCertRef` is currently non-optional in the `AliasConfiguration`. If I put them both in `AliasConfiguration`, I will have to make them optional.

Are you also okay with deprecating `aliasConfiguration` instead, and allowing to keep both `acmCertRef` and `acmCert` required? If yes, I'd a appreciate a suggestion for the new `alias` property. 👍